src/game/server/swarm/asw_marine.cpp

```diff
@@ -5063,6 +5063,12 @@ void CASW_Marine::ASW_Ignite( float flFlameLifetime, float flSize, CBaseEntity *
 
 	bool bFriendlyFire = IRelationType( pAttacker ) == D_LI;
 
+	// special scaling for flamer ff
+	if (asw_marine_ff_absorption.GetInt() > 0 && bFriendlyFire && pDamagingWeapon->Classify() == CLASS_ASW_FLAMER)
+	{
+		flFlameLifetime = MAX( 1.0f, flFlameLifetime * m_fFriendlyFireAbsorptionTime );
```

Comment:
I recommend making the 1.0 a convar so it's easier to change the minimum burn time for testing and for challenges.